From 8f4d9ccbb4c0c83ef77ee37d68fff18b14a01e3f Mon Sep 17 00:00:00 2001
From: Peter Swain <pswain@cavium.com>
Date: Mon, 21 Jul 2014 18:05:50 -0700
Subject: [PATCH 676/974] MIPS: octeon-ethernet: no need for sentinel pool

commit 1a842aeaa cleaned up hard-coded CVMX_FPA1_NUM_POOLS,
but added an unnecessary instance at end. Drop it.
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/net/ethernet/octeon/ethernet-mem.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/octeon/ethernet-mem.c b/drivers/net/ethernet/octeon/ethernet-mem.c
index dbab7b3..d069167 100644
--- a/drivers/net/ethernet/octeon/ethernet-mem.c
+++ b/drivers/net/ethernet/octeon/ethernet-mem.c
@@ -46,8 +46,8 @@ struct fpa_pool {
 };
 
 static DEFINE_MUTEX(cvm_oct_pools_lock);
-static struct fpa_pool cvm_oct_pools[CVMX_FPA1_NUM_POOLS+1] = {
-	[0 ... CVMX_FPA1_NUM_POOLS] = {-1,},
+static struct fpa_pool cvm_oct_pools[CVMX_FPA1_NUM_POOLS] = {
+	[0 ... CVMX_FPA1_NUM_POOLS - 1] = {-1,},
 };
 
 /**
-- 
2.6.2

